crates/matrix-sdk-ui/src/room_list_service/mod.rs

```diff
@@ -395,7 +395,14 @@ impl RoomListService {
             settings.required_state.push((StateEventType::RoomCreate, "".to_owned()));
         }
 
-        self.sliding_sync.subscribe_to_rooms(room_ids, Some(settings))
+        let cancel_in_flight_request = match self.state.get() {
+            State::Init | State::Recovering | State::Error { .. } | State::Terminated { .. } => {
+                false
+            }
+            State::SettingUp | State::Running => true,
```

Comment:
> turn this new argument to true if the state machine isn't in a “starting” state.

This seems to set it to `true` when we're in the `Running` state, are you sure this is correct? `State::Init` sounds more like it would fit the description.

Reply:
It is correct yes. `Init` and `Recovering` are the “starting” states, thus we don't want to cancel in-flight requests. `SettingUp` and `Running` are states where we have a `pos`, so we can cancel in-flight requests here.

The state machine is like this: Init -> SettingUp -> Running. In case of an error, it depends on the previous state before the error, but basically if we were in `Running`, we jump back to Recovering -> Running. So Init and Recovering are basically the “starting” states here.